unitstatushandler/softwaremanager.go

```diff
@@ -244,8 +250,10 @@ func (manager *softwareManager) processDesiredStatus(desiredStatus cloudprotocol
 
 	manager.processDesiredLayers(update, allLayers, desiredStatus.Layers)
 
+	needRunInstances := manager.processRunInstances(desiredStatus.Instances)
+
 	if len(update.DownloadServices) != 0 || len(update.RemoveServices) != 0 ||
-		len(update.DownloadLayers) != 0 || len(update.RemoveLayers) != 0 {
+		len(update.DownloadLayers) != 0 || len(update.RemoveLayers) != 0 || needRunInstances {
```

Comment:
Also you need to check new update if one is in progress. See line 731:
```go
reflect.DeepEqual(update.RunInstance, manager.CurrentUpdate.RunInstances)
```
should be added.